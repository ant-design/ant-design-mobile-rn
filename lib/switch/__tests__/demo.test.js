import rnDemoTest from '../../../tests/shared/demoTest';

rnDemoTest('switch');
