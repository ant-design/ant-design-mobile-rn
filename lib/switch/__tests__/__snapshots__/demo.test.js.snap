// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/switch/demo/basic.tsx correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderTopColor": "#dddddd",
        "borderTopWidth": 0.5,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "flexGrow": 1,
              "paddingLeft": 15,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 44,
                "paddingRight": 15,
                "paddingVertical": 6,
              },
              false,
              false,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 17,
                    "textAlignVertical": "center",
                  },
                ]
              }
            >
              On(controlled)
            </Text>
          </View>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#4dd865"
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            value={true}
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "flexGrow": 1,
              "paddingLeft": 15,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 44,
                "paddingRight": 15,
                "paddingVertical": 6,
              },
              false,
              false,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 17,
                    "textAlignVertical": "center",
                  },
                ]
              }
            >
              Off(controlled)
            </Text>
          </View>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#4dd865"
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "flexGrow": 1,
              "paddingLeft": 15,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 44,
                "paddingRight": 15,
                "paddingVertical": 6,
              },
              false,
              false,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 17,
                    "textAlignVertical": "center",
                  },
                ]
              }
            >
              onChange event, switch status: 
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 17,
                    "textAlignVertical": "center",
                  },
                ]
              }
            >
              close
            </Text>
          </View>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#4dd865"
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "flexGrow": 1,
              "paddingLeft": 15,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 44,
                "paddingRight": 15,
                "paddingVertical": 6,
              },
              false,
              false,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 17,
                    "textAlignVertical": "center",
                  },
                ]
              }
            >
              disabled
            </Text>
          </View>
          <RCTSwitch
            disabled={true}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#4dd865"
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "flexGrow": 1,
              "paddingLeft": 15,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 44,
                "paddingRight": 15,
                "paddingVertical": 6,
              },
              false,
              false,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 17,
                    "textAlignVertical": "center",
                  },
                ]
              }
            >
              color
            </Text>
          </View>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="red"
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            value={true}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
        ]
      }
    />
  </View>
</View>
`;
